resources/utilities/upload_vcf.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+
+"""
+Usage: $0 PATIENT_ID VCF CREDENTIALS_FILE SERVER_NAME
+
+Upload the VCF file as an attachment for the given patient. For example:
+
+    $0 P0000102 path/to/file.vcf path/to/credentials.txt server_name
+
+credentials.txt should have one line with the form:
+
+    username:password
+"""
+
+from __future__ import with_statement
+
+import sys
+import os
+import logging
+import json
+
+try:
+    from httplib import HTTPConnection
+except ImportError:
+    from http.client import HTTPSConnection
+try:
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urlencode
+from base64 import b64encode
+import xml.etree.ElementTree as ET
+
+# Set globally after parsing arguments
+CREDENTIALS = None
+SERVER = None
+
+def load_credentials(filename):
+    global CREDENTIALS
+    logging.info('Using credentials in file: {0}'.format(filename))
+    with open(filename) as ifp:
+        CREDENTIALS = ifp.read().strip()
+
+class Request(object):
+    def __init__(self, method, resource, content_type='text/plain', **kwargs):
+        auth = b64encode(CREDENTIALS.encode()).decode()
+        self._headers = {
+            'Authorization': 'Basic {0}'.format(auth),
+            'Content-Type': content_type,
+            'Accept': '*/*'
+            }
+        self._method = method
+        self._href = resource
+        self._kwargs = kwargs
+        self._conn = None
+
+    def __enter__(self):
+        self._conn = HTTPSConnection(SERVER)
+        #self._conn.set_debuglevel(1)
+        self._conn.request(self._method, self._href, headers=self._headers, **self._kwargs)
+        response = self._conn.getresponse()
+        return response
+
+    def __exit__(self, *args, **kwargs):
+        if self._conn:
+            self._conn.close()
+
+def get_consent(record_id):
+    logging.info('Loading Consents for record: {0}'.format(record_id))
+    href = '/rest/patients/{0}/consents/'.format(record_id)
+    result = 'no'
```

Comment:
This requires that the genetic consent is enabled. Change this to "yes" so that it still works in the absence of the genetic consent.